### ISSUE TITLE
Wrong documentation of credential usage [API-1212]

### ISSUE THREAD

**Issue by emreyigit** [hazelcast/hazelcast-csharp-client#580]
Alternative credential options are documented wrong. 

The [documentation ](http://hazelcast.github.io/hazelcast-csharp-client/latest/doc/configuration/options.html#a-nameauthentication-authentication) explains alternative authentication options. However, `username-password`, `kerberos` and `token` are not working as documented. In order to use these options, parameters should be provided under `args` field which will be parsed to `Dictionary<string,string>` to pass ctor of related credential provider.

Implementation of `InjectionOptions` only deals with `typeName` and `args`. Other custom fields are supported in `args`.

Usage should be corrected as 

``` json
"username-password":
{
    "args":{
         "username": "someone",
         "password": "secret"
    }
}


"kerberos":
{
  "args":{
      "spn": "service-provider-name"
  }
}


"token":
{
   "args":{
        "data": "some-secret-token",
        "encoding": "none"
   }
}

```

**Comment by zpqrtbnk** [comment]
That is a bug. The *purpose* is to work as documented. Let me check.

**Comment by zpqrtbnk** [comment]
OK so this is... mysterious. Like, a file never got merged or something. My mistake, quite probably, but still I cannot explain it.

The documented syntax is the one we want to have because it's the "nice" one:

```json
"authentication": {
    "username-password": {
        "username": "bob",
        "password": "secret"
    }
}
```

And the "args" indirection is there only because... it's a leftover of some intermediate state where these options were "injected options" where you specify a type name + some args for the constructor. So the binding of these options, as it is today in the code, makes plainly no sense. I have attached a PR to this issue which should correct the situation.

Let me know what you think.
